### PR TITLE
refactor: RegionEngine::handle_request always returns affected rows

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This closes https://github.com/GreptimeTeam/greptimedb/issues/2784.

@waynexia I wonder if we should add a type alias or wrapper for `AffectedRows` but I keep use `usize` in this patch for a first step.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @Dysprosium0626 you can take a review or take over this patch as the assignee. But it happens that I have time to take a look at this issue and it's weeks after your assignment.

File: src/mito2/src/request.rs [hunk 1]
```diff
@@ -384,16 +382,16 @@ pub(crate) fn validate_proto_value(
 
 /// Oneshot output result sender.
 #[derive(Debug)]
-pub(crate) struct OutputTx(Sender<Result<Output>>);
```
Comment (tisonkun) [inline src/mito2/src/request.rs:387]: For the comment in PR description, it's mainly if we need the `AffectedRows` name in these type references.

**Comment by WenyXu** [comment]
I prefer using words to align the DQL, DML, DDL, and DCL. The definition of `execution` is too broad.

**Comment by tisonkun** [comment]
@WenyXu do you mean revert the name to `handle_request` but retain the return value type refactor?

**Comment by tisonkun** [comment]
The reason I try to rename is `request` is actually non-query request but perhaps `execution` is not optimal also.

**Comment by WenyXu** [comment]
> @WenyXu do you mean revert the name to `handle_request` but retain the return value type refactor?

Yes, the `execution ` seems not good either. Maybe it's a better time to rename the method when we need to separate handle_request into several methods.

**Comment by tisonkun** [comment]
@WenyXu Updated.

**Comment by evenyag** [comment]
Shall we add a `RegionResponse` struct?
```rust
struct RegionResponse {
    affected_rows: usize,
}
```

**Comment by Dysprosium0626** [comment]
> cc @Dysprosium0626 you can take a review or take over this patch as the assignee. But it happens that I have time to take a look at this issue and it's weeks after your assignment.

@tisonkun I am sorry that I was busy on other things last week. I had done it halfway but delayed because I had to take some time to understand the code structure (I'm new here). Anyway, I have the similar idea as your refraction and thank you for taking over my task. I will take some other tasks.

**Comment by tisonkun** [comment]
> @waynexia I wonder if we should add a type alias or wrapper for AffectedRows but I keep use usize in this patch for a first step.

@evenyag I thought of it either. But lean to keep `usize` until we found the return value as `usize` causes confusion.

**Comment by waynexia** [comment]
> > @waynexia I wonder if we should add a type alias or wrapper for AffectedRows but I keep use usize in this patch for a first step.
> 
> @evenyag I thought of it either. But lean to keep `usize` until we found the return value as `usize` causes confusion.

Thanks for your efforts.

For the return type, I prefer at least a type alias over plain `usize`, which makes it easier to distinguish with other `usize`s

**Comment by tisonkun** [comment]
OK. I'll push a new commit later :D

**Comment by tisonkun** [comment]
@WenyXu @evenyag @waynexia updated.

BTW we have three type alias `pub type AffectedRows = usize;`. Perhaps we can move them to a common base crate as a follow-up.

**Review by evenyag — APPROVED** [review]

File: src/metric-engine/src/engine.rs [hunk 2]
```diff
@@ -108,30 +108,24 @@ impl RegionEngine for MetricEngine {
...
         let result = match request {
             RegionRequest::Put(put) => self.inner.put_region(region_id, put).await,
             RegionRequest::Delete(_) => todo!(),
-            RegionRequest::Create(create) => self
-                .inner
-                .create_region(region_id, create)
-                .await
-                .map(|_| Output::AffectedRows(0)),
+            RegionRequest::Create(create) => {
+                self.inner.create_region(region_id, create).await.map(|_| 0)
```
Comment (waynexia) [inline src/metric-engine/src/engine.rs:121]: Can you also change the underlying method's return type?

File: src/metric-engine/src/engine.rs [hunk 3]
```diff
@@ -108,30 +108,24 @@ impl RegionEngine for MetricEngine {
...
             RegionRequest::Drop(_) => todo!(),
             RegionRequest::Open(_) => todo!(),
             RegionRequest::Close(_) => todo!(),
-            RegionRequest::Alter(alter) => self
-                .inner
-                .alter_region(region_id, alter)
-                .await
-                .map(|_| Output::AffectedRows(0)),
+            RegionRequest::Alter(alter) => {
+                self.inner.alter_region(region_id, alter).await.map(|_| 0)
```
Comment (waynexia) [inline src/metric-engine/src/engine.rs:127]: Ditto

**Review by waynexia — APPROVED** [review]
Almost LGTM, thanks for this @tisonkun

**Comment by tisonkun** [comment]
@waynexia Updated at https://github.com/GreptimeTeam/greptimedb/pull/2874/commits/911a4c1e5a21b87dbf178e77eb520dc5bdfba31a

**Pull request merged**